core/src/main/scala/chisel3/experimental/hierarchy/core/Instance.scala

```diff
@@ -107,9 +108,33 @@ object Instance extends SourceInfoDoc {
     implicit sourceInfo: SourceInfo,
     compileOptions:      CompileOptions
   ): Instance[T] = {
+    // Check to see if the module is already defined internally or externally
+    val existingMod = Builder.components.map {
+      case c: DefModule if c.id == definition.proto          => Some(c)
+      case c: DefBlackBox if c.name == definition.proto.name => Some(c)
+      case _ => None
+    }.flatten
```

Comment:
Jack previously pointed out that this is a linear search. Would it be worthwhile to change `Builder.components` from an `ArrayBuffer` to a `HashMap` for better lookup speed?

Reply:
I don't think its worthwhile unless you can measure the speedup; your lookup function would be complicated because you'd have to find the `DefBlackBox` case anyways.